stl/inc/flat_set

```diff
@@ -430,27 +438,25 @@ public:
     }
 
 private:
-    void _Assert_after_sorted_input() const {
-        _STL_ASSERT(_STD is_sorted(cbegin(), cend(), _Get_comp_v()), "Input was not sorted!");
-        if constexpr (!_Multi) {
-            _STL_ASSERT(_Is_unique(), "Input was sorted but not unique!");
-        }
-    }
-
-    bool _Is_unique() const {
-        if (empty()) {
-            return true;
-        }
-        const const_iterator _End = cend();
-        const_iterator _It        = cbegin();
-        while (++_It != _End) {
-            if (_Keys_equal(*(_It - 1), *_It)) {
-                return false;
+    bool _Check_sorted(const_iterator _It, const const_iterator _End) const {
+        if (_Multi) {
+            return _STD is_sorted(_It, _End, _Get_comp_v());
+        } else {
+            // sorted-unique
+            if (_It == _End) {
+                return true;
             }
+            while (++_It != _End) {
+                if (!_Compare(*(_It - 1), *_It)) {
+                    return false;
+                }
+            }
+            return true;
         }
-        return true;
     }
 
+    static constexpr const char* _Msg_not_sorted = _Multi ? "Input was not sorted!" : "Input was not sorted-unique!";

```

Comment:
I have no idea how to represent this gracefully :( It might be helpful to an extra `_STL_VERIFY` that repeat error message just like `_STL_INTERNAL_CHECK` and `assert`.

Also, these assertion messages are currently not correct as `_Base_flat_set` is also the base for `flat_multiset`.
https://github.com/microsoft/STL/blob/6bedc29de49c653314a4d8b100c4f20bb50b6069/stl/inc/flat_set#L47-L49
https://github.com/microsoft/STL/blob/6bedc29de49c653314a4d8b100c4f20bb50b6069/stl/inc/flat_set#L65-L66